app/src/main/java/org/novinger/jason/demoreader/ArticleListActivity.java

```diff
@@ -1,72 +1,54 @@
 package org.novinger.jason.demoreader;
 
 import android.os.StrictMode;
+import android.support.annotation.NonNull;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.util.Log;
 import android.view.Menu;
 import android.view.MenuItem;
-import android.view.View;
-import android.view.ViewPropertyAnimator;
-import android.widget.AdapterView;
-import android.widget.ArrayAdapter;
-import android.widget.ListView;
 
 import com.google.gson.Gson;
 
 import org.novinger.jason.demoreader.datamodels.Article;
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.Response;
 
-public class ListActivity extends AppCompatActivity {
+public class ArticleListActivity extends AppCompatActivity {
 
-    private final static int qty = 50;
+    private final static int qty = 250;
     private final static String feedUrl = "http://www.dailydot.com/api/v1/content/article/";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_list);
 
-        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
-        StrictMode.setThreadPolicy(policy);
-        Article[] articles;
-        try {
-            articles = getFeedItems(getFeedUrl());
-        } catch (IOException ex) {
-            articles = new Article[0];
-            Log.d("Exception", ex.toString());
-        }
+        enableNetworkingPolicy();
 
-        final ListView listview = (ListView) findViewById(R.id.listview);
-        final ArrayList<String> list = new ArrayList<>();
-        for (Article article : articles) {
-            list.add(article.toString());
-        }
+        final ArrayList<Article> list = getArticles();
```

Comment:
Hrm, good question. I probably got excited there about providing optimization hints to the compiler. Since it's a local variable that doesn't get reassigned, it should not cause any issues, I believe. The `list` object is still mutable, the label just isn't reassignable.